### PR TITLE
Corr comp

### PR DESCRIPTION
**Pull request by yidawang**
Add a high-performance correlation computation function using BLAS wrapped in Cython to `brainiak/fcma/util`. Running at least one order of magnitude faster than `np.corrcoef`.
```
Ours: 100 loops, best of 3: 14.2 ms per loop
np.corrcoef: 10 loops, best of 3: 153 ms per loop
```

### CODE REVIEW

File: brainiak/fcma/util.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    return data
+
+
+def compute_correlation(set1, set2):
+    """compute correlation between two sets
+
+    Correlate the rows of set1 with the rows of set2.
+    If set1 == set2, it is auto-correlation computation
+    resulting in a symmetric correlation matrix.
+    The number of columns MUST agree between set1 and set2
```
Comment (mihaic) [inline brainiak/fcma/util.py]: Could you please add a reference for the method of correlation computation you are using? Some Latex code could help too.

File: brainiak/fcma/util.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    shape = data.shape
+    data = zscore(data, axis=axis, ddof=0)
+    # if zscore fails (standard deviation is zero),
+    # set all values to be zero
+    data = np.nan_to_num(data)
+    data = data / math.sqrt(shape[axis])
+    return data
+
+
+def compute_correlation(set1, set2):
```
Comment (mihaic) [inline brainiak/fcma/util.py]: Shouldn't the parameters be called `array1/2` or `matrix1/2`?

Comment (yidawang) [inline brainiak/fcma/util.py]: they are essentially two matrices, depicting two sets of variables. I can use `matrix1/2`

File: brainiak/fcma/util.py (hunk 1 above)
Comment (yidawang) [inline brainiak/fcma/util.py]: done

File: brainiak/fcma/util.py [hunk 3]
```diff
@@ -70,6 +70,19 @@ def compute_correlation(matrix1, matrix2):
 
     is the standard deviation of variable X
 
+    Reducing the correlation computation to matrix multiplication
+    and use BLAS GEMM API wrapped by Scipy can speedup the numpy build-in
```
Comment (mihaic) [inline brainiak/fcma/util.py]: use->using, build-in->built-in

Comment (yidawang) [inline brainiak/fcma/util.py]: done

File: brainiak/fcma/util.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+# Authors: Yida Wang
+# (Intel Labs), 2017
+
+import numpy as np
+from . import cython_blas as blas
+from scipy.stats.mstats import zscore
+import math
+
+
+def normalize_for_correlation(data, axis):
```
Comment (mihaic) [inline brainiak/fcma/util.py]: Is it necessary for this function to be public?

Comment (yidawang) [inline brainiak/fcma/util.py]: May or may not be used by others in the future. But for now, let me make it private

**Pull request merged**